// angular-simple-scroll-spy/__ivy_ngcc__/fesm2015/angular-simple-scroll-spy.js
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output, Renderer2, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
class ScrollSpyContentDirective {
    /**
     * @param {?} _element
     */
    constructor(_element) {
        this._element = _element;
        this.sscCurrentContentChange = new EventEmitter();
        this.sscDirection = "row";
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onScroll(event) {
        /** @type {?} */
        let nowContent = Array.from(this._element.nativeElement.children)
            .filter(child => this.target === child.tagName)
            .reverse()
            .find(child => this.sscDirection === "row" ?
            (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
            (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft);
        if (!nowContent)
            return;
        if (nowContent.id !== this._currentContent) {
            this._currentContent = nowContent.id;
            this.sscCurrentContentChange.emit(this._currentContent);
        }
    }
}
ScrollSpyContentDirective.ɵfac = function ScrollSpyContentDirective_Factory(t) { return new (t || ScrollSpyContentDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ScrollSpyContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ScrollSpyContentDirective, selectors: [["", "scrollSpyContent", ""]], hostBindings: function ScrollSpyContentDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("scroll", function ScrollSpyContentDirective_scroll_HostBindingHandler($event) { return ctx.onScroll($event); });
    } }, inputs: { sscDirection: "sscDirection", target: ["scrollSpyContent", "target"], sscCurrentContent: "sscCurrentContent" }, outputs: { sscCurrentContentChange: "sscCurrentContentChange" } });
/** @nocollapse */
ScrollSpyContentDirective.ctorParameters = () => [
    { type: ElementRef }
];
ScrollSpyContentDirective.propDecorators = {
    target: [{ type: Input, args: ['scrollSpyContent',] }],
    sscCurrentContent: [{ type: Input }],
    sscCurrentContentChange: [{ type: Output }],
    sscDirection: [{ type: Input }],
    onScroll: [{ type: HostListener, args: ['scroll', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ScrollSpyContentDirective, [{
        type: Directive,
        args: [{
                selector: '[scrollSpyContent]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { sscCurrentContentChange: [{
            type: Output
        }], sscDirection: [{
            type: Input
        }], onScroll: [{
            type: HostListener,
            args: ['scroll', ['$event']]
        }], target: [{
            type: Input,
            args: ['scrollSpyContent']
        }], sscCurrentContent: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class ScrollSpyMenuDirective {
    /**
     * @param {?} _element
     * @param {?} _renderer
     */
    constructor(_element, _renderer) {
        this._element = _element;
        this._renderer = _renderer;
        this.ssmActiveClassName = "active";
        this.ssmScrollBehavior = "smooth";
        this.ssmScrollBlock = "start";
        this.ssmScrollInline = "nearest";
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set ssmCurrentContent(value) {
        if (this.contentId === value) {
            this._renderer.addClass(this._element.nativeElement, this.ssmActiveClassName);
        }
        else {
            this._renderer.removeClass(this._element.nativeElement, this.ssmActiveClassName);
        }
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onClick(event) {
        document.querySelector(`#${this.contentId}`).scrollIntoView({
            behavior: this.ssmScrollBehavior,
            block: this.ssmScrollBlock,
            inline: this.ssmScrollInline
        });
    }
}
ScrollSpyMenuDirective.ɵfac = function ScrollSpyMenuDirective_Factory(t) { return new (t || ScrollSpyMenuDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
ScrollSpyMenuDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ScrollSpyMenuDirective, selectors: [["", "scrollSpyMenu", ""]], hostBindings: function ScrollSpyMenuDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ScrollSpyMenuDirective_click_HostBindingHandler($event) { return ctx.onClick($event); });
    } }, inputs: { ssmActiveClassName: "ssmActiveClassName", ssmScrollBehavior: "ssmScrollBehavior", ssmScrollBlock: "ssmScrollBlock", ssmScrollInline: "ssmScrollInline", ssmCurrentContent: "ssmCurrentContent", contentId: ["scrollSpyMenu", "contentId"] } });
/** @nocollapse */
ScrollSpyMenuDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ScrollSpyMenuDirective.propDecorators = {
    contentId: [{ type: Input, args: ['scrollSpyMenu',] }],
    ssmActiveClassName: [{ type: Input }],
    ssmCurrentContent: [{ type: Input }],
    ssmScrollBehavior: [{ type: Input }],
    ssmScrollBlock: [{ type: Input }],
    ssmScrollInline: [{ type: Input }],
    onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ScrollSpyMenuDirective, [{
        type: Directive,
        args: [{
                selector: '[scrollSpyMenu]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }]; }, { ssmActiveClassName: [{
            type: Input
        }], ssmScrollBehavior: [{
            type: Input
        }], ssmScrollBlock: [{
            type: Input
        }], ssmScrollInline: [{
            type: Input
        }], ssmCurrentContent: [{
            type: Input
        }], onClick: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], contentId: [{
            type: Input,
            args: ['scrollSpyMenu']
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class SimpleScrollSpyModule {
}
SimpleScrollSpyModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SimpleScrollSpyModule });
SimpleScrollSpyModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SimpleScrollSpyModule_Factory(t) { return new (t || SimpleScrollSpyModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SimpleScrollSpyModule, { declarations: [ScrollSpyContentDirective,
        ScrollSpyMenuDirective], exports: [ScrollSpyContentDirective,
        ScrollSpyMenuDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SimpleScrollSpyModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [
                    ScrollSpyContentDirective,
                    ScrollSpyMenuDirective
                ],
                exports: [
                    ScrollSpyContentDirective,
                    ScrollSpyMenuDirective
                ]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { SimpleScrollSpyModule, ScrollSpyContentDirective as ɵa, ScrollSpyMenuDirective as ɵb };
