// angular-simple-scroll-spy/bundles/angular-simple-scroll-spy.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('angular-simple-scroll-spy', ['exports', '@angular/core'], factory) :
    (factory((global['angular-simple-scroll-spy'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var ScrollSpyContentDirective = (function () {
        function ScrollSpyContentDirective(_element) {
            this._element = _element;
            this.sscCurrentContentChange = new core.EventEmitter();
            this.sscDirection = "row";
        }
        /**
         * @param {?} event
         * @return {?}
         */
        ScrollSpyContentDirective.prototype.onScroll = /**
         * @param {?} event
         * @return {?}
         */
            function (event) {
                var _this = this;
                /** @type {?} */
                var nowContent = Array.from(this._element.nativeElement.children)
                    .filter(function (child) { return _this.target === child.tagName; })
                    .reverse()
                    .find(function (child) {
                    return _this.sscDirection === "row" ?
                        (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
                        (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft;
                });
                if (!nowContent)
                    return;
                if (nowContent.id !== this._currentContent) {
                    this._currentContent = nowContent.id;
                    this.sscCurrentContentChange.emit(this._currentContent);
                }
            };
        ScrollSpyContentDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[scrollSpyContent]'
                    },] },
        ];
        /** @nocollapse */
        ScrollSpyContentDirective.ctorParameters = function () {
            return [
                { type: core.ElementRef }
            ];
        };
        ScrollSpyContentDirective.propDecorators = {
            target: [{ type: core.Input, args: ['scrollSpyContent',] }],
            sscCurrentContent: [{ type: core.Input }],
            sscCurrentContentChange: [{ type: core.Output }],
            sscDirection: [{ type: core.Input }],
            onScroll: [{ type: core.HostListener, args: ['scroll', ['$event'],] }]
        };
        return ScrollSpyContentDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var ScrollSpyMenuDirective = (function () {
        function ScrollSpyMenuDirective(_element, _renderer) {
            this._element = _element;
            this._renderer = _renderer;
            this.ssmActiveClassName = "active";
            this.ssmScrollBehavior = "smooth";
            this.ssmScrollBlock = "start";
            this.ssmScrollInline = "nearest";
        }
        Object.defineProperty(ScrollSpyMenuDirective.prototype, "ssmCurrentContent", {
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                if (this.contentId === value) {
                    this._renderer.addClass(this._element.nativeElement, this.ssmActiveClassName);
                }
                else {
                    this._renderer.removeClass(this._element.nativeElement, this.ssmActiveClassName);
                }
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @param {?} event
         * @return {?}
         */
        ScrollSpyMenuDirective.prototype.onClick = /**
         * @param {?} event
         * @return {?}
         */
            function (event) {
                document.querySelector("#" + this.contentId).scrollIntoView({
                    behavior: this.ssmScrollBehavior,
                    block: this.ssmScrollBlock,
                    inline: this.ssmScrollInline
                });
            };
        ScrollSpyMenuDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[scrollSpyMenu]'
                    },] },
        ];
        /** @nocollapse */
        ScrollSpyMenuDirective.ctorParameters = function () {
            return [
                { type: core.ElementRef },
                { type: core.Renderer2 }
            ];
        };
        ScrollSpyMenuDirective.propDecorators = {
            contentId: [{ type: core.Input, args: ['scrollSpyMenu',] }],
            ssmActiveClassName: [{ type: core.Input }],
            ssmCurrentContent: [{ type: core.Input }],
            ssmScrollBehavior: [{ type: core.Input }],
            ssmScrollBlock: [{ type: core.Input }],
            ssmScrollInline: [{ type: core.Input }],
            onClick: [{ type: core.HostListener, args: ['click', ['$event'],] }]
        };
        return ScrollSpyMenuDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var SimpleScrollSpyModule = (function () {
        function SimpleScrollSpyModule() {
        }
        SimpleScrollSpyModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [],
                        declarations: [
                            ScrollSpyContentDirective,
                            ScrollSpyMenuDirective
                        ],
                        exports: [
                            ScrollSpyContentDirective,
                            ScrollSpyMenuDirective
                        ]
                    },] },
        ];
        return SimpleScrollSpyModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.SimpleScrollSpyModule = SimpleScrollSpyModule;
    exports.ɵa = ScrollSpyContentDirective;
    exports.ɵb = ScrollSpyMenuDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
