// angular-simple-scroll-spy/fesm2015/angular-simple-scroll-spy.js
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output, Renderer2, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class ScrollSpyContentDirective {
    /**
     * @param {?} _element
     */
    constructor(_element) {
        this._element = _element;
        this.sscCurrentContentChange = new EventEmitter();
        this.sscDirection = "row";
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onScroll(event) {
        /** @type {?} */
        let nowContent = Array.from(this._element.nativeElement.children)
            .filter(child => this.target === child.tagName)
            .reverse()
            .find(child => this.sscDirection === "row" ?
            (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
            (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft);
        if (!nowContent)
            return;
        if (nowContent.id !== this._currentContent) {
            this._currentContent = nowContent.id;
            this.sscCurrentContentChange.emit(this._currentContent);
        }
    }
}
ScrollSpyContentDirective.decorators = [
    { type: Directive, args: [{
                selector: '[scrollSpyContent]'
            },] },
];
/** @nocollapse */
ScrollSpyContentDirective.ctorParameters = () => [
    { type: ElementRef }
];
ScrollSpyContentDirective.propDecorators = {
    target: [{ type: Input, args: ['scrollSpyContent',] }],
    sscCurrentContent: [{ type: Input }],
    sscCurrentContentChange: [{ type: Output }],
    sscDirection: [{ type: Input }],
    onScroll: [{ type: HostListener, args: ['scroll', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class ScrollSpyMenuDirective {
    /**
     * @param {?} _element
     * @param {?} _renderer
     */
    constructor(_element, _renderer) {
        this._element = _element;
        this._renderer = _renderer;
        this.ssmActiveClassName = "active";
        this.ssmScrollBehavior = "smooth";
        this.ssmScrollBlock = "start";
        this.ssmScrollInline = "nearest";
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set ssmCurrentContent(value) {
        if (this.contentId === value) {
            this._renderer.addClass(this._element.nativeElement, this.ssmActiveClassName);
        }
        else {
            this._renderer.removeClass(this._element.nativeElement, this.ssmActiveClassName);
        }
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onClick(event) {
        document.querySelector(`#${this.contentId}`).scrollIntoView({
            behavior: this.ssmScrollBehavior,
            block: this.ssmScrollBlock,
            inline: this.ssmScrollInline
        });
    }
}
ScrollSpyMenuDirective.decorators = [
    { type: Directive, args: [{
                selector: '[scrollSpyMenu]'
            },] },
];
/** @nocollapse */
ScrollSpyMenuDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ScrollSpyMenuDirective.propDecorators = {
    contentId: [{ type: Input, args: ['scrollSpyMenu',] }],
    ssmActiveClassName: [{ type: Input }],
    ssmCurrentContent: [{ type: Input }],
    ssmScrollBehavior: [{ type: Input }],
    ssmScrollBlock: [{ type: Input }],
    ssmScrollInline: [{ type: Input }],
    onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class SimpleScrollSpyModule {
}
SimpleScrollSpyModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [
                    ScrollSpyContentDirective,
                    ScrollSpyMenuDirective
                ],
                exports: [
                    ScrollSpyContentDirective,
                    ScrollSpyMenuDirective
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { SimpleScrollSpyModule, ScrollSpyContentDirective as ɵa, ScrollSpyMenuDirective as ɵb };
