// ng-google-one-tap/esm2015/lib/ng-google-one-tap.service.js
import { Injectable, NgZone } from '@angular/core';
import { Inject } from '@angular/core';
import { CONFIGURATION } from './toke';
import { Subject } from 'rxjs';
export class NgOneTapService {
    constructor(envConfig, ngZone) {
        this.envConfig = envConfig;
        this.ngZone = ngZone;
        this._promtMoment = new Subject();
        this._oneTapCredentialResponse = new Subject();
        this._authUserResponse = new Subject();
    }
    get promtMoment() {
        return this._promtMoment.asObservable();
    }
    get oneTapCredentialResponse() {
        return this._oneTapCredentialResponse.asObservable();
    }
    get authUserResponse() {
        return this._authUserResponse.asObservable();
    }
    tapInitialize(config) {
        window.onload = () => {
            this.tapRender(config);
        };
        if (document.readyState == 'complete') {
            this.tapRender(config);
        }
    }
    tapRender(config) {
        const conf = Object.assign(Object.assign({}, this.envConfig), config);
        if (!!(conf === null || conf === void 0 ? void 0 : conf.disable_exponential_cooldowntime)) {
            document.cookie = 'g_state' + '=;Path=/;';
        }
        window.google.accounts.id.initialize(Object.assign(Object.assign({}, conf), { callback: (auth) => {
                this.ngZone.run(() => {
                    if (!!conf.authvalidate_by_googleapis) {
                        const http = new XMLHttpRequest();
                        const url = `https://oauth2.googleapis.com/tokeninfo?id_token=${auth.credential}`;
                        http.open('GET', url);
                        http.send();
                        http.onreadystatechange = () => {
                            if (http.readyState == 4 && http.status == 200) {
                                this._authUserResponse.next(JSON.parse(http.responseText));
                            }
                        };
                    }
                    this._oneTapCredentialResponse.next(auth);
                });
            } }));
        window.google.accounts.id.prompt((pmt) => {
            this._promtMoment.next(pmt);
        });
    }
    signOut() {
        window.google.accounts.id.disableAutoSelect();
    }
    cancelTheTap() {
        window.google.accounts.id.cancel();
    }
}
NgOneTapService.decorators = [
    { type: Injectable }
];
NgOneTapService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [CONFIGURATION,] }] },
    { type: NgZone }
];
