// angular-simple-scroll-spy/esm2015/lib/simple-scroll-spy-content.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output } from '@angular/core';
export class ScrollSpyContentDirective {
    /**
     * @param {?} _element
     */
    constructor(_element) {
        this._element = _element;
        this.sscCurrentContentChange = new EventEmitter();
        this.sscDirection = "row";
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onScroll(event) {
        /** @type {?} */
        let nowContent = Array.from(this._element.nativeElement.children)
            .filter(child => this.target === child.tagName)
            .reverse()
            .find(child => this.sscDirection === "row" ?
            (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
            (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft);
        if (!nowContent)
            return;
        if (nowContent.id !== this._currentContent) {
            this._currentContent = nowContent.id;
            this.sscCurrentContentChange.emit(this._currentContent);
        }
    }
}
ScrollSpyContentDirective.decorators = [
    { type: Directive, args: [{
                selector: '[scrollSpyContent]'
            },] },
];
/** @nocollapse */
ScrollSpyContentDirective.ctorParameters = () => [
    { type: ElementRef }
];
ScrollSpyContentDirective.propDecorators = {
    target: [{ type: Input, args: ['scrollSpyContent',] }],
    sscCurrentContent: [{ type: Input }],
    sscCurrentContentChange: [{ type: Output }],
    sscDirection: [{ type: Input }],
    onScroll: [{ type: HostListener, args: ['scroll', ['$event'],] }]
};
if (false) {
    /** @type {?} */
    ScrollSpyContentDirective.prototype.target;
    /** @type {?} */
    ScrollSpyContentDirective.prototype._currentContent;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscCurrentContent;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscCurrentContentChange;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscDirection;
    /** @type {?} */
    ScrollSpyContentDirective.prototype._element;
}
