// angular-simple-scroll-spy/esm5/lib/simple-scroll-spy-menu.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
var ScrollSpyMenuDirective = /** @class */ (function () {
    function ScrollSpyMenuDirective(_element, _renderer) {
        this._element = _element;
        this._renderer = _renderer;
        this.ssmActiveClassName = "active";
        this.ssmScrollBehavior = "smooth";
        this.ssmScrollBlock = "start";
        this.ssmScrollInline = "nearest";
    }
    Object.defineProperty(ScrollSpyMenuDirective.prototype, "ssmCurrentContent", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            if (this.contentId === value) {
                this._renderer.addClass(this._element.nativeElement, this.ssmActiveClassName);
            }
            else {
                this._renderer.removeClass(this._element.nativeElement, this.ssmActiveClassName);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    ScrollSpyMenuDirective.prototype.onClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        document.querySelector("#" + this.contentId).scrollIntoView({
            behavior: this.ssmScrollBehavior,
            block: this.ssmScrollBlock,
            inline: this.ssmScrollInline
        });
    };
    ScrollSpyMenuDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[scrollSpyMenu]'
                },] },
    ];
    /** @nocollapse */
    ScrollSpyMenuDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    ScrollSpyMenuDirective.propDecorators = {
        contentId: [{ type: Input, args: ['scrollSpyMenu',] }],
        ssmActiveClassName: [{ type: Input }],
        ssmCurrentContent: [{ type: Input }],
        ssmScrollBehavior: [{ type: Input }],
        ssmScrollBlock: [{ type: Input }],
        ssmScrollInline: [{ type: Input }],
        onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
    };
    return ScrollSpyMenuDirective;
}());
export { ScrollSpyMenuDirective };
if (false) {
    /** @type {?} */
    ScrollSpyMenuDirective.prototype.contentId;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype.ssmActiveClassName;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype.ssmScrollBehavior;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype.ssmScrollBlock;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype.ssmScrollInline;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype._element;
    /** @type {?} */
    ScrollSpyMenuDirective.prototype._renderer;
}
