// angular-simple-scroll-spy/esm5/lib/simple-scroll-spy-content.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output } from '@angular/core';
var ScrollSpyContentDirective = /** @class */ (function () {
    function ScrollSpyContentDirective(_element) {
        this._element = _element;
        this.sscCurrentContentChange = new EventEmitter();
        this.sscDirection = "row";
    }
    /**
     * @param {?} event
     * @return {?}
     */
    ScrollSpyContentDirective.prototype.onScroll = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        var _this = this;
        /** @type {?} */
        var nowContent = Array.from(this._element.nativeElement.children)
            .filter(function (child) { return _this.target === child.tagName; })
            .reverse()
            .find(function (child) { return _this.sscDirection === "row" ?
            (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
            (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft; });
        if (!nowContent)
            return;
        if (nowContent.id !== this._currentContent) {
            this._currentContent = nowContent.id;
            this.sscCurrentContentChange.emit(this._currentContent);
        }
    };
    ScrollSpyContentDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[scrollSpyContent]'
                },] },
    ];
    /** @nocollapse */
    ScrollSpyContentDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ScrollSpyContentDirective.propDecorators = {
        target: [{ type: Input, args: ['scrollSpyContent',] }],
        sscCurrentContent: [{ type: Input }],
        sscCurrentContentChange: [{ type: Output }],
        sscDirection: [{ type: Input }],
        onScroll: [{ type: HostListener, args: ['scroll', ['$event'],] }]
    };
    return ScrollSpyContentDirective;
}());
export { ScrollSpyContentDirective };
if (false) {
    /** @type {?} */
    ScrollSpyContentDirective.prototype.target;
    /** @type {?} */
    ScrollSpyContentDirective.prototype._currentContent;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscCurrentContent;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscCurrentContentChange;
    /** @type {?} */
    ScrollSpyContentDirective.prototype.sscDirection;
    /** @type {?} */
    ScrollSpyContentDirective.prototype._element;
}
