// angular-simple-scroll-spy/fesm5/angular-simple-scroll-spy.js
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output, Renderer2, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var ScrollSpyContentDirective = /** @class */ (function () {
    function ScrollSpyContentDirective(_element) {
        this._element = _element;
        this.sscCurrentContentChange = new EventEmitter();
        this.sscDirection = "row";
    }
    /**
     * @param {?} event
     * @return {?}
     */
    ScrollSpyContentDirective.prototype.onScroll = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        var _this = this;
        /** @type {?} */
        var nowContent = Array.from(this._element.nativeElement.children)
            .filter(function (child) { return _this.target === child.tagName; })
            .reverse()
            .find(function (child) { return _this.sscDirection === "row" ?
            (child.offsetTop - event.target.offsetTop) <= event.target.scrollTop :
            (child.offsetLeft - event.target.offsetLeft) <= event.target.scrollLeft; });
        if (!nowContent)
            return;
        if (nowContent.id !== this._currentContent) {
            this._currentContent = nowContent.id;
            this.sscCurrentContentChange.emit(this._currentContent);
        }
    };
    ScrollSpyContentDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[scrollSpyContent]'
                },] },
    ];
    /** @nocollapse */
    ScrollSpyContentDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ScrollSpyContentDirective.propDecorators = {
        target: [{ type: Input, args: ['scrollSpyContent',] }],
        sscCurrentContent: [{ type: Input }],
        sscCurrentContentChange: [{ type: Output }],
        sscDirection: [{ type: Input }],
        onScroll: [{ type: HostListener, args: ['scroll', ['$event'],] }]
    };
    return ScrollSpyContentDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var ScrollSpyMenuDirective = /** @class */ (function () {
    function ScrollSpyMenuDirective(_element, _renderer) {
        this._element = _element;
        this._renderer = _renderer;
        this.ssmActiveClassName = "active";
        this.ssmScrollBehavior = "smooth";
        this.ssmScrollBlock = "start";
        this.ssmScrollInline = "nearest";
    }
    Object.defineProperty(ScrollSpyMenuDirective.prototype, "ssmCurrentContent", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            if (this.contentId === value) {
                this._renderer.addClass(this._element.nativeElement, this.ssmActiveClassName);
            }
            else {
                this._renderer.removeClass(this._element.nativeElement, this.ssmActiveClassName);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    ScrollSpyMenuDirective.prototype.onClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        document.querySelector("#" + this.contentId).scrollIntoView({
            behavior: this.ssmScrollBehavior,
            block: this.ssmScrollBlock,
            inline: this.ssmScrollInline
        });
    };
    ScrollSpyMenuDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[scrollSpyMenu]'
                },] },
    ];
    /** @nocollapse */
    ScrollSpyMenuDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    ScrollSpyMenuDirective.propDecorators = {
        contentId: [{ type: Input, args: ['scrollSpyMenu',] }],
        ssmActiveClassName: [{ type: Input }],
        ssmCurrentContent: [{ type: Input }],
        ssmScrollBehavior: [{ type: Input }],
        ssmScrollBlock: [{ type: Input }],
        ssmScrollInline: [{ type: Input }],
        onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
    };
    return ScrollSpyMenuDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var SimpleScrollSpyModule = /** @class */ (function () {
    function SimpleScrollSpyModule() {
    }
    SimpleScrollSpyModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [
                        ScrollSpyContentDirective,
                        ScrollSpyMenuDirective
                    ],
                    exports: [
                        ScrollSpyContentDirective,
                        ScrollSpyMenuDirective
                    ]
                },] },
    ];
    return SimpleScrollSpyModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { SimpleScrollSpyModule, ScrollSpyContentDirective as ɵa, ScrollSpyMenuDirective as ɵb };
